### PR TITLE
Add Debian Bullseye ansible-test image

### PR DESCRIPTION
**Pull request by felixfontein**

### CODE REVIEW

**Comment by dmsimard** [comment]
@felixfontein WFM on fedora 35 with:
```bash
> rpm -qa |egrep "buildah|podman"
podman-gvproxy-3.4.4-1.fc35.x86_64
podman-plugins-3.4.4-1.fc35.x86_64
podman-3.4.4-1.fc35.x86_64
buildah-1.23.1-1.fc35.x86_64
```

File: ansible-test/debian-bullseye/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+coverage==4.5.4
+resolvelib==0.5.4
```
Comment (dmsimard) [inline ansible-test/debian-bullseye/requirements.txt:2]:
```suggestion
resolvelib==0.5.4
# include ara for optional test and CI reporting
ara
```

File: ansible-test/debian-bullseye/build.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Builds a centos-stream8 image suitable for use with ansible-test
```
Comment (dmsimard) [inline ansible-test/debian-bullseye/build.sh:2]:
```suggestion
# Builds a debian bullseye image suitable for use with ansible-test
```

File: ansible-test/debian-bullseye/build.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+# Extra python dependencies
+buildah run --volume ${SCRIPT_DIR}:/tmp/src:z "${build}" -- /bin/bash -c "pip3 install -r /tmp/src/requirements.txt"
+
+# Ansible-specific setup:
+buildah run "${build}" -- /bin/bash -c "ln -s /lib/systemd/systemd /sbin/init"
+buildah run "${build}" -- /bin/bash -c "rm /etc/apt/apt.conf.d/docker-clean"
+buildah run "${build}" -- /bin/bash -c "mkdir /etc/ansible/"
+buildah run "${build}" -- /bin/bash -c "/bin/echo -e \"[local]\nlocalhost ansible_connection=local\" > /etc/ansible/hosts"
+buildah run "${build}" -- /bin/bash -c "locale-gen en_US.UTF-8"
```
Comment (dmsimard) [inline ansible-test/debian-bullseye/build.sh:19]: We have an opportunity to use the same commands as the [centos one](https://github.com/ansible-community/images/blob/8ead278761bfbf15ffa4c28e01fc7f0c5a3d8b1b/ansible-test/centos-stream8/build.sh#L19-L20):

```suggestion
# Ansible-specific setup:
buildah run "${build}" -- /bin/bash -c "ln -s /lib/systemd/systemd /sbin/init"
buildah run "${build}" -- /bin/bash -c "rm /etc/apt/apt.conf.d/docker-clean"
buildah run "${build}" -- /bin/bash -c "ssh-keygen -A && sed -i -e 's/^\(Defaults\s*requiretty\)/#--- \1/' /etc/sudoers"
buildah run "${build}" -- /bin/bash -c "mkdir -p /etc/ansible && echo -e '[local]\nlocalhost ansible_connection=local' > /etc/ansible/hosts"
buildah run "${build}" -- /bin/bash -c "locale-gen en_US.UTF-8"
```

File: ansible-test/debian-bullseye/dependencies.txt [hunk 4]
```diff
@@ -0,0 +1,51 @@
+acl
+apache2
```
Comment (dmsimard) [inline ansible-test/debian-bullseye/dependencies.txt:2]: ``apache2`` ?

We don't have ``httpd`` in the centos image.
Do you think it's necessary ?

**Comment by dmsimard** [comment]
I have built and uploaded this PR to https://quay.io/ansible-community/test-image:debian-bullseye so that at least it exists. It can be updated automatically later.

File: ansible-test/debian-bullseye/dependencies.txt (hunk 4 above)
Comment (felixfontein) [reply ansible-test/debian-bullseye/dependencies.txt:2]: There are quite a few packages in here where I wonder whether they are really necessary. I took the list from the Ubuntu 20.04 image. My *guess* is that some useful CLI program is part of the `apache` package on Debian/Ubuntu... But let's just remove it. If it turns out there's something useful missing, we can still add it later.

**Comment by felixfontein** [comment]
> Not a big deal for now but I thought I would mention that quay.io reports a sizeable amount of vulnerabilities in the image (if it is accurate): https://quay.io/repository/ansible-community/test-image/manifest/sha256:1cdf2a404f0e3207939a22a2d1772dc2153de35db8e098d26c2640c7a986e2cf?tab=vulnerabilities

Looking at the corresponding Debian bug tracker issues for these vulnerabilities, these have been intentionally not fixed by Debian (for various reasons). I don't think these should be a problem, also since the intention for this test image is to do tests with in a typical Debian environment, where these issues will also be present (if the appropriate packages are installed).

**Comment by felixfontein** [comment]
ready_for_review

**Comment by dmsimard** [comment]
@felixfontein could you rebase on top of the main branch ? I merged the archlinux PR.

File: .github/workflows/ansible-test.yml [hunk 5]
```diff
@@ -38,9 +40,15 @@ jobs:
           ${{ matrix.name }}/build.sh
         working-directory: ./ansible-test/
 
+      # Right now it is not possible to use some of the images (Debian Bullseye, ArchLinux) with podman in GitHub Actions,
+      # while others (CentOS 8 Stream) work fine. This is not specific to the images in this repository, the official
+      # ansible-test Ubuntu 20.04 image also does not work with podman in GHA, but does work with Docker in GHA.
```
Comment (dmsimard) [inline .github/workflows/ansible-test.yml:45]: I created an issue so we don't forget about it: https://github.com/ansible-community/images/issues/9

**Comment by felixfontein** [comment]
I've rebased the PR. In both README and GHA workflow, I ordered the images by alphabet.

**Comment by felixfontein** [comment]
@dmsimard thanks for reviewing and merging!

**Pull request merged**